geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -804,58 +810,22 @@ def default_input_types(self) -> list[str]:
         """Choice of survey creation types."""
         return self.__INPUT_TYPE
 
-    @property
-    def tx_id_property(self) -> ReferencedData | IntegerData | None:
-        """
-        Data link between the receiver and transmitter object.
-        """
-        if self._tx_id_property is None and self.metadata is not None:
-            self._tx_id_property = self._fetch_transmitter_id()
-
-        return self._tx_id_property
-
-    @tx_id_property.setter
-    def tx_id_property(self, value: uuid.UUID | ReferencedData | np.ndarray | None):
-        if isinstance(value, uuid.UUID):
-            value = self.get_data(value)[0]
-
-        if isinstance(value, np.ndarray):
-            attributes = {
-                "values": value.astype(np.int32),
+    def _format_transmitter_ids(self, values, attributes):
```

Comment:
this function could be in BaseEMSurvey as you are using it for tipper too?
Just changing the first word of the value_map

Reply:
Main difference is that LargeLoop and Tipper are referenced data with specific labels. Hence my thought of having slightly different methods on each.